### PR TITLE
The traditional but updated hello world builder

### PR DESCRIPTION
**Pull request by rsandell**

### CODE REVIEW

File: hello-world/src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            </excludes>
+        </fileSet>
+        <fileSet filtered="true" packaged="true">
+            <directory>src/main/java</directory>
+        </fileSet>
+        <fileSet filtered="true" packaged="true">
+            <directory>src/test/java</directory>
+        </fileSet>
+    </fileSets>
+</archetype-descriptor>
```
Comment (jglick) [inline hello-world/src/main/resources/META-INF/maven/archetype-metadata.xml]: Suggest

```xml
    <requiredProperties>
        <requiredProperty key="groupId">
            <defaultValue>unused</defaultValue>
        </requiredProperty>
        <requiredProperty key="package">
            <defaultValue>org.jenkinsci.plugins.sample</defaultValue>
        </requiredProperty>
    </requiredProperties>
```

File: hello-world/pom.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.jenkins.archetypes</groupId>
+        <artifactId>archetypes-parent</artifactId>
+        <version>1.1-SNAPSHOT</version>
+    </parent>
+    <groupId>io.jenkins.archetypes</groupId>
+    <artifactId>hello-world-plugin</artifactId>
+    <name>Hello World Builder Jenkins Plugin</name>
```
Comment (jglick) [inline hello-world/pom.xml]: `<description>` as in #6

File: hello-world/src/main/resources/archetype-resources/pom.xml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    <properties>
+        <jenkins.version>2.7.3</jenkins.version>
+        <workflow-step-api-plugin.version>2.12</workflow-step-api-plugin.version>
+        <workflow-support-plugin.version>2.14</workflow-support-plugin.version>
+        <workflow-cps-plugin.version>2.39</workflow-cps-plugin.version>
+    </properties>
+    <name>TODO Plugin</name>
+    <description>TODO</description>
+    <url>https://wiki.jenkins-ci.org/display/JENKINS/TODO+Plugin</url>
+    <licenses>
```
Comment (jglick) [inline hello-world/src/main/resources/archetype-resources/pom.xml]: As in https://github.com/jenkinsci/maven-hpi-plugin/pull/74 you should offer a comment here I suppose.

File: hello-world/src/main/resources/archetype-resources/pom.xml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            <groupId>org.jenkins-ci.plugins.workflow</groupId>
+            <artifactId>workflow-cps</artifactId>
+            <version>${workflow-cps-plugin.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins.workflow</groupId>
+            <artifactId>workflow-cps</artifactId>
+            <version>${workflow-cps-plugin.version}</version>
+            <classifier>tests</classifier>
```
Comment (jglick) [inline hello-world/src/main/resources/archetype-resources/pom.xml]: Unused after you removed `SnippetizerTester`.

File: pom.xml [hunk 5]
```diff
@@ -24,6 +24,7 @@
     <modules>
         <module>empty-plugin</module>
         <module>scripted-pipeline</module>
+        <module>hello-world</module>
```
Comment (jglick) [inline pom.xml]: move up one line for clarity

File: hello-world/src/main/resources/archetype-resources/pom.xml (hunk 3 above)
Comment (jglick) [reply hello-world/src/main/resources/archetype-resources/pom.xml]: Similarly commented-out sections for `developers`, `scm`, and explanation of `jenkins.version`.

File: hello-world/src/main/resources/META-INF/maven/archetype-metadata.xml (hunk 1 above)
Comment (rsandell) [reply hello-world/src/main/resources/META-INF/maven/archetype-metadata.xml]: done

File: hello-world/pom.xml (hunk 2 above)
Comment (rsandell) [reply hello-world/pom.xml]: done

File: hello-world/src/main/resources/archetype-resources/pom.xml (hunk 3 above)
Comment (rsandell) [reply hello-world/src/main/resources/archetype-resources/pom.xml]: done

File: hello-world/src/main/resources/archetype-resources/pom.xml (hunk 4 above)
Comment (rsandell) [reply hello-world/src/main/resources/archetype-resources/pom.xml]: done

File: pom.xml (hunk 5 above)
Comment (rsandell) [reply pom.xml]: done

**Comment by christ66** [comment]
@rsandell What happened to all of the Javadocs from the original hello world builder? I think it's important to keep the previous javadoc information as someone who's building a hello-world app should have some documentation to learn more.

**Comment by rsandell** [comment]
@christ66 I copied from jesse's talk so maybe stuff got lost from there.
You are welcome to add all the docs you can muster :)

**Pull request merged**